scipy/stats/tests/test_morestats.py

```diff
@@ -1564,6 +1564,53 @@ def test_gh_6873(self):
         # to only about five significant digits.
         assert_allclose(lam, -0.051654, rtol=1e-5)
 
+    @pytest.mark.parametrize("bounds", [(-1, 1), (0, 1), (-2, -1)])
+    def test_bounded_optimizer_within_bounds(self, bounds):
+        # Define custom optimizer with bounds.
+        def optimizer(fun, args):
+            return optimize.minimize_scalar(fun, bounds=bounds, args=args,
+                                            method="bounded")
+
+        _, lmbda = stats.boxcox(_boxcox_data, lmbda=None, optimizer=optimizer)
+        assert bounds[0] < lmbda < bounds[1]
+
+    def test_bounded_optimizer_against_unbounded_optimizer(self):
+        # Test whether setting bounds on optimizer excludes solution from unbounded
+        # optimizer.
+
+        # Get unbounded solution.
+        _, lmbda = stats.boxcox(_boxcox_data, lmbda=None)
+
+        # Set bounds around solution.
+        bounds = (lmbda - 1, lmbda + 1)
```

Comment:
We wanted an interval that excluded the original solution `lmbda`. As it is:
```
lmbda
Out[12]: -0.051654323802509894

lmbda_bounded
Out[13]: -0.051654220523443525
```
Which aren't exactly equal, but might as well be, so they don't really test that this does what it's supposed to do.

Here, we should set bounds to, say, 
```
bounds = (lmbda + 0.1, lmbda + 1)
```
Assuming there are no local minima within the interval (etc.), we would then expect that new solution would be the lower of these bounds.

I'll change this in the PR to your branch.

Along these lines, I'll also make a small tweak to your `test_bounded_optimizer_within_bounds` that makes it a little stronger.

Reply:
Haha, yes, my bounds didn't exactly exclude the solution! Good catch.